src/functions-and-arrays.js

```diff
@@ -26,13 +51,29 @@ function sum() {}
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(numbersAvg) {
+
+  let sum=0;
+for (let i=0; i<numbersAvg.length;i++){
+  sum+= numbersAvg[i];
+}
+return sum/ numbersAvg.length;
+}
 
 
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
-function averageWordLength() { }
+function averageWordLength(wordsArr) {
+
+ let sum =0;
+  for (let i=0; i<arr.length;i++){
+    sum+=wordsArr[i].length;
+  }
+  return sum /  wordsArr.length;
+}
+
+
```

Comment:
- Aquí solo te ha faltado cubrir el caso de que si el array está vacío, la función retorne null y que en la línea 70 has llamado a `arr.length`! Pero en tu caso tu array se llama `wordsArr`, por lo tanto no lo reconoce:

```suggestion
function averageWordLength(wordsArr) {

 if (wordsArr.length ===  0) { return null }

 let sum =0;
  for (let i=0; i<wordsArr.length;i++){
    sum+=wordsArr[i].length;
  }
  return sum /  wordsArr.length;
}
```